packages/stream-metadata/src/handleHealthCheckRequest.ts

```diff
@@ -0,0 +1,32 @@
+import { FastifyReply, FastifyRequest } from 'fastify'
+
+import { Config } from './types'
+import { getLogger } from './logger'
+import { getRiverRegistry } from './evmRpcClient'
+
+const logger = getLogger('handleHealthCheckRequest')
+
+export async function handleHealthCheckRequest(
+	config: Config,
+	request: FastifyRequest,
+	reply: FastifyReply,
+) {
+	let riverRegistry: ReturnType<typeof getRiverRegistry> | undefined
+	// Do a health check on the river registry
+	try {
+		riverRegistry = getRiverRegistry(config)
+		if (riverRegistry) {
+			await riverRegistry.getAllNodes()
+			// healthy
+			return reply.code(200).send({ status: 'ok' })
+		}
+
+		// unhealthy
+		logger.error('Failed to get river registry')
+		return reply.code(500).send({ status: 'error' })
```

Comment:
@tak-hntlabs the `@river-build/web3` library's `getWeb3Deployment` function takes care of getting the river registry, amongst many other things. i think having multiple paths that instantiate a river registry is harmless at best, and confusing at worst. it's an abstraction that we should probably remove. if you think you need to maintain the ability to create different river registries for whatever reason - which i personally think we don't but feel free to correct me - you can still use the `getWeb3Deployment` function.

i also have a nit here: riverRegistry can be initialized as a `const` within the try block

Reply:
Let's do a screenshare code walkthrough.  `getWeb3Deployment` just returns config info like the riverRegistry address. For health check, I'm making a contract call to the river registry using the @river/web3 RiverRegistry shim. The original code was using the `viem` package function `getContract`. Removed and replaced with `RiverRegistry` shim.